frontend/apps/web/app/(mgmt)/[account]/new/connection/postgres/PostgresForm.tsx

```diff
@@ -130,6 +136,66 @@ export default function PostgresForm() {
     }
   }
 
+  /* we call the underlying useGetConnection API directly since we can't call
+the hook in the useEffect conditionally. 
+*/
+  useEffect(() => {
```

Comment:
I hate useEffects, but overall I think this makes sense here given how we want to do this.

I'd like to see this method cleaned up a bit though prior to merge to improve its readability.

Generally when we make requests we pull them out into a function to abstract away all of the data handling. tons of examples of that around the app. Should clean this up considerably.

Reply:
Yeah totally agree, cleaned it up a bit.